### DISCUSSION TITLE
TypeError: 'MoleculeDatapoint' object is not iterable

### DISCUSSION THREAD

**Discussion by ankitadu** [chemprop/chemprop#1119] [Q&A]
Hello, 
I am trying to use chemprop to analyse an assay activity dataset. I am using the following Jyputer notebook example as guidance:  https://chemprop.readthedocs.io/en/latest/training.html.
I keep running into the above error, which appears after the following lines of code:
featurizer = featurizers.SimpleMoleculeMolGraphFeaturizer()
train_dset = data.MoleculeDataset(train_data[0], featurizer)
scaler = train_dset.normalize_targets()
val_dset = data.MoleculeDataset(val_data[0], featurizer)
val_dset.normalize_targets(scaler)
test_dset = data.MoleculeDataset(test_data[0], featurizer)

As far as I can tell my data structure is fine and its similar to some classicifation training data supplied with the chemprop installation. Specifically I am refereing to the follwoing dataset:  tests/data/classification/mol.csv. So, the model is being trained on multiple columns in this case. 
I have tried to train with the data set supplied with chemprop as a test case as well as with my own data set. I always run into this error. 
So far I have tried to define 'MoleculeDatapoint' as an itteratable but have had no success. Can I please get some help/insight on how I can debug this issue?
All steps prior to the featurization step seem to be working perfectly fine. Happy to provide more info if needed!

Thanks in advance 
Ankita

**Comment by JacksonBurns** [comment]
Are you running with chemprop version 2.1?

**Reply by ankitadu** [reply]
Yes as far as I can tell I am. I recently downloaded it after the release of the version 2. Is there a way to make sure?

**Reply by JacksonBurns** [reply]
If you can run `pip list` from where ever you have it installed we can see the exact version

**Reply by ankitadu** [reply]
Thanks for your response. I can see that the chemprop version I have running is 2.0.4 and not 2.1. 
Could this be a known bug and should I try to re-install chemprop? 
Additionally, what would be the best way to re-install? In the past I just cloned the github repository.

**Reply by JacksonBurns** [reply]
This is not a bug, the demo notebooks are just intended to work with the latest version. You will need to update, syntax for splitting data has changed.

If you previously installed by cloning, just `git pull` to get the latest changes and then install.

I would suggest instead though that you just install with pip, i.e. `pip install chemprop`.

**Reply by ankitadu** [reply]
Update: I installed chemprop version 2.1. However, the issue persists. I wonder if this might be because of invalid SMILES being present in my dataset? Or perhaps Chemprop does not recognize a SMILE for example something like: [As]#[In] ?

**Reply by JacksonBurns** [reply]
Could you post a more complete code example of what you are running, as well as the full stacktrace?

**Reply by ankitadu** [reply]
Sure, please see attached screenshots
![Screenshot (28)](https://github.com/user-attachments/assets/02fde7d1-c7a7-4139-b967-3a72b79060df)
![Screenshot (29)](https://github.com/user-attachments/assets/6cb0e1f7-e495-406f-84f0-515383744a3a)
![Screenshot (30)](https://github.com/user-attachments/assets/034248f9-9cd9-4ede-a37f-b0ff419d0564)
![Screenshot (32)](https://github.com/user-attachments/assets/f05b5794-b115-4acc-8e26-51e49e934d17)

Also just for confirmation: as far as I can tell I am running chemprop 2.1 
![Screenshot (33)](https://github.com/user-attachments/assets/9d703306-20af-43d6-b829-a98364923cac)

**Comment by JacksonBurns** [comment]
Can you post the actual code or attach the notebook itself so I can execute it?

**Reply by ankitadu** [reply]
Following up on this again!
Since posting this issue I have tried to validate and standardize the list of SMILES of interest and my data seems to pass all the checks.
Quite confused because I can varify that chemprop is running prefectly well with the example data supplied with the chemprop installation. However when I try the same process with my own data I run into the issue mentioned above.

The code I am using can be found here:
https://github.com/ankitadu/Chemprop

Happy to provide any further information needed. Would appriciate any further insight on this!

**Reply by KnathanM** [reply]
I downloaded your notebook and this cell ran without error:
```
featurizer = featurizers.SimpleMoleculeMolGraphFeaturizer()

train_dset = data.MoleculeDataset(train_data[0], featurizer)
val_dset = data.MoleculeDataset(val_data[0], featurizer)
test_dset = data.MoleculeDataset(test_data[0], featurizer)
```

I was using a chemprop install from github with the latest version of main.

**Comment by ankitadu** [comment]
Apologies for any confusion. Please find the relevant files in the follwoing repo:
https://github.com/ankitadu/Chemprop

Here I have tried to train on both an ensamble of data from different assays as well as on single assay data. Both apporaches lead to the same error.
Hope this is better. Please let me know if I can provide any further information
